File: bindings/python/examples/quickstart.py
import sys

import vietasr


def main(wav_path: str) -> None:
    pipeline = vietasr.Pipeline.preset("transcribe")
    result = pipeline.transcribe(wav_path)
    print(result.text)


if __name__ == "__main__":
    main(sys.argv[1] if len(sys.argv) > 1 else "sample.wav")

File: bindings/nodejs/examples/quickstart.js
const path = require("path");
const vietasr = require("..");

function main() {
    const wavPath = process.argv[2];
    if (!wavPath) {
        console.error(`usage: node ${path.basename(__filename)} <wav-file>`);
        console.error("\nAvailable presets:", vietasr.listPresets());
        console.error("Available modules:", vietasr.listModules());
        process.exit(1);
    }

    const pipe = vietasr.Pipeline.preset("transcribe");
    const result = pipe.transcribe(wavPath);
    console.log(result.text);
    pipe.close();
}

main();
